resources/js/src/app/components/basket/AddItemToBasketOverlay.js

```diff
@@ -68,11 +68,35 @@ Vue.component("add-item-to-basket-overlay", {
         {
             if (this.basketItem.currentBasketItem.calculatedPrices)
             {
-                this.price = this.basketItem.currentBasketItem.calculatedPrices.default.price;
+                this.price = this.basketItem.currentBasketItem.calculatedPrices.default.price + this.calculateSurcharge();
                 this.currency = this.basketItem.currentBasketItem.calculatedPrices.default.currency;
             }
         },
 
+        calculateSurcharge()
+        {
+
+            let sumSurcharge = 0;
+
+            for (const property of this.basketItem.currentBasketItem.properties)
+            {
+
+                if (typeof property.property.value !== "undefined" && property.property.value.length > 0)
+                {
+                    if (property.surcharge > 0)
+                    {
+                        sumSurcharge += property.surcharge;
+                    }
+                    else if (property.property.surcharge > 0)
+                    {
+                        sumSurcharge += property.property.surcharge;
+                    }
```

Comment:
if (property.surcharge > 0)
 +                    {
 +                        sumSurcharge += property.surcharge;
 +                    }
 +                    else if (property.property.surcharge > 0)
 +                    {
 +                        sumSurcharge += property.property.surcharge;
 +                    }

wieso prüfst du in dem if auf property? Reicht es nicht allgemein auf property.property zu prüfen?